apm-agent-plugins/apm-profiling-plugin/src/main/java/co/elastic/apm/agent/profiler/asyncprofiler/AsyncProfiler.java

```diff
@@ -71,6 +73,8 @@ public static AsyncProfiler getInstance(String profilerLibDirectory) {
                         "profiling_inferred_spans_enabled to false");
                 }
                 try {
+                    // set the AsyncProfiler.safemode with the configured safemode, so that optimizations can be applied already at load time
+                    System.setProperty(SAFEMODE_SYSTEM_PROPERTY_NAME, String.valueOf(safemode));
```

Comment:
Can the safemode still be set to a different value at runtime?
Also, which optimizations?

Reply:
I will add comments.
The new System property is going to be removed in 2.x, where `CompiledMethodLoad` are not going to be enabled at all (which also answers the "optimization" question.

Reply:
https://github.com/elastic/apm-agent-java/pull/2165/commits/3f35ae1f7b69ebccabba0bc7083f8ba36552a648